respy/pre_processing/data_checking.py

```diff
@@ -17,7 +19,9 @@ def check_estimation_data(df, optim_paras):
         If data has not the expected format.
 
     """
-    df = df.reset_index()
+    # Make sure all columns are available.
+    labels, _ = generate_column_labels_estimation(optim_paras)
+    df = df[labels[2:]].reset_index()
```

Comment:
Would generate column labels simulation be better here ?

Reply:
As before. I will rename it. Closing this one and keep the other one.